simple-dns/src/dns/rdata/caa.rs

```diff
@@ -0,0 +1,79 @@
+use crate::dns::{CharacterString, PacketPart};
+
+use super::RR;
+
+// RFC 8659: Allow domain name holders to indicate whether they are authorized to issue digital certificates for particular domain name
+// Used as a security policy for certificate authorities
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct CAA<'a> {
```

Comment:
we should make a comment similar to the NSAP one that says that this doesn't validate that the CharacterString is a valid value, only that the bytes are split properly. 

There are reasons we would want to create an invalid one, so I think we should allow people to do that, but I think we should follow their pattern and warn them.